src/polydb.jl

```diff
@@ -0,0 +1,284 @@
+   return BSONCursor(Mongoc.find(c.mcol, Mongoc.BSON(d); options=opts))
+end
+
+function find(c::BSONCollection, d::Pair...)
+   return BSONCursor(Mongoc.find(c.mcol, Mongoc.BSON(d...)))
+end
+
+# creating `BSON` iterators from the respective `Polymake.BigObject` iterator
+function BSONCollection(c::Collection)
+   return BSONCollection(c.mcol)
+end
+
+function BSONCursor(cursor::Cursor)
+   return BSONCursor(cursor.mcursor)
+end
+
+# returns a Polymake.BigObject from a Mongoc.BSON document
+function parse_document(bson::Mongoc.BSON)
+   str = Mongoc.as_json(bson)
+   return call_function(:common, :deserialize_json_string, str)
+end
+
+#Iterator
+
+# help functions to reduce runtime and keep code clean
+function _evaluateIteration(nothing::Nothing)
+   return nothing
+end
+
+function _evaluateIteration(a::Tuple{Mongoc.BSON, Any})
+   return (parse_document(a[1]), a[2])
+end
+
+# default iteration functions returning `Polymake.BigObject`s
+function Base.iterate(cursor::Cursor)
+   return _evaluateIteration(iterate(cursor.mcursor))
+end
+
+function Base.iterate(cursor::Cursor, state::Nothing)
+   return _evaluateIteration(iterate(cursor.mcursor, state))
+end
+
+function Base.collect(cursor::Cursor)
+   result = Vector{Polymake.BigObject}()
+    for doc in cursor
+        push!(result, doc)
+    end
+    return result
+end
+
+function Base.iterate(coll::Collection)
+   return _evaluateIteration(iterate(coll.mcol))
+end
+
+function Base.iterate(coll::Collection, state::Mongoc.Cursor)
+   return _evaluateIteration(iterate(coll.mcol, state))
+end
+
+# functions for `BSON` iteration
+function Base.iterate(cursor::BSONCursor)
+   return iterate(cursor.mcursor)
+end
+
+function Base.iterate(cursor::BSONCursor, state::Nothing)
+   return iterate(cursor.mcursor, state)
+end
+
+function Base.collect(cursor::BSONCursor)
+   return collect(cursor.mcursor)
+end
+
+function Base.iterate(coll::BSONCollection)
+   return iterate(coll.mcol)
+end
+
+function Base.iterate(coll::BSONCollection, state::Mongoc.Cursor)
+   return iterate(coll.mcol, state)
+end
+
+#Info
+
+# prints information about a specific Collection
+# also used for the info(::Database) function
+function info(coll::Collection)
+   db = coll.mcol.database
+   coll_c = db[string("_collectionInfo.", coll.mcol.name)]
+   info = iterate(coll_c)[1]
+   _print_collection(info)
+end
+
+# returns an array containing the names of all collections in the Polydb
+function _get_collection_names(db::Database)
+   opts = Mongoc.BSON("authorizedCollections" => true, "nameOnly" => true)
+   return Mongoc.get_collection_names(db.mdb;options=opts)
+end
+
+# for the set of names obtained by the _get_collection_names(::Database) function
+# returns two arrays containing the names of the meta data collections
+# first one for sections, second one for collections
+function _get_meta_names(names::Array{String, 1})
+   n = length(names)
+   sec_bool = Array{Bool, 1}(undef, n)
```

Comment:
do we need `Array{Bool, 1}`, or would `BitVector` be enough? also please specify (for clarity) that this is `Base.Array` and not `Polymake.Array`